### DISCUSSION TITLE
Run ArgoCD under Docker?

### DISCUSSION THREAD

**Discussion by john2exonets** [argoproj/argo-cd#11596] [Q&A]
Is there a way to run ArgoCD under Docker?  Looking to try it out without having to take up a K8s Cluster.  I don't need a production deployment, just something for a Lab.

There is no documentation (that I can find) about running it under Docker.  Just doing a simple 'run' command, the ArgoCD container just continuously restarts.

**Comment by emirot** [comment]
I don't think they have a docker compose, but what you could to is run it locally with Kind or enable k8s in docker-desktop settings.

**Comment by pedroteixeira** [comment]
Maybe this one here answers this question: https://hub.docker.com/r/bitnami/argo-cd/ - but it might be just the CLI tool under docker

**Comment by jannfis** [comment]
Argo CD consists of multiple components that need to run in parallel in order for it to function correctly. You basically can run each of these components in their own container, as long as the components can communicate with each other and are set up correctly. However, this potentially becomes pretty tedious to set up.

My recommendation would be to go with a small, single-node and self-contained Kubernetes distribution such as kind, k3s or microk8s and just install Argo CD to that cluster. Otherwise, if you have VirtualBox and vagrant, you can try out [Argo CD in a nutshell](https://github.com/jannfis/argocd-nutshell).
